sdlib/d/gc/extent.d

```diff
@@ -338,8 +350,75 @@ public:
 			return 0;
 		}
 
-		// Decode freespace, found in the final byte (or two bytes) of the alloc:
-		return readPackedFreeSpace(cast(ushort*) (address + slotSize - 2));
+		// Decode freespace, found in the final usable byte (or two bytes) :
+		return readPackedFreeSpace(
+			cast(ushort*) (address + getUsableSpace(index) - 2));
+	}
+
+	size_t getUsableSpace(uint index) {
```

Comment:
Not only that shouldn't really be necessary, but this is wrong. This is the second time incorrect code is in there and passes tests. So I'll let add tests for boundary condition and break this down until it is obvious none of these remain. Relying on me to find all the problem is not a solution.